src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1325,6 +1325,23 @@ public synchronized void close() throws InterruptedException {
     }
 
     /**
+     * Close this client object as the {@link #close() } method.
+     * This method will wait for internal resources to be released.
+     *
+     * @param waitForShutdownTimeoutMs timeout (in milliseconds) to wait for resources to be released.
+     * Use zero or a negative value to skip the wait
+     * @throws InterruptedException
+     * @return true if waitForShutdownTimeout is greater than zero and all of the resources have been released
+     */
+    public boolean close(int waitForShutdownTimeoutMs) throws InterruptedException {
+        close();
+        if (waitForShutdownTimeoutMs > 0) {
```

Comment:
I don't think this check is needed - actually, it sounds better to remove this check so a negative value will trigger illegal argument exception indicates something is wrong in upstream code, because it's hard to know if the negative value is passed on purpose, or is an accident.